src/metrics.ts

```diff
@@ -0,0 +1,378 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as grpc from '@grpc/grpc-js';
+import { Registry, Counter, Histogram, metric } from 'prom-client';
+import { Code, connectErrorFromReason, Interceptor, StreamRequest, UnaryRequest } from '@bufbuild/connect-node';
+import { MethodKind } from '@bufbuild/protobuf';
+import { codeToString, StreamResponse, UnaryResponse } from '@bufbuild/connect-core';
+
+export type GrpcMethodType = 'unary' | 'client_stream' | 'server_stream' | 'bidi_stream';
+
+export interface IGrpcCallMetricsLabels {
+    service: string;
+    method: string;
+    type: GrpcMethodType;
+}
+
+export interface IGrpcCallMetricsLabelsWithCode extends IGrpcCallMetricsLabels {
+    code: string;
+}
+
+const register = new Registry();
+
+class PrometheusClientCallMetrics {
+    readonly startedCounter: Counter<string>;
+    readonly sentCounter: Counter<string>;
+    readonly receivedCounter: Counter<string>;
+    readonly handledCounter: Counter<string>;
+    readonly handledSecondsHistogram: Histogram<string>;
+
+    constructor() {
+        this.startedCounter = new Counter({
+            name: 'grpc_client_started_total',
+            help: 'Total number of RPCs started on the client.',
+            labelNames: ['grpc_service', 'grpc_method', 'grpc_type'],
+            registers: [register],
+        });
+        this.sentCounter = new Counter({
```

Comment:
We don't have `grpc_client_msg_sent_total`, `grpc_client_msg_received_total` metrics added in ide-metrics whitelist and don't need them